app/controllers/api/v3/public/channels_controller.rb

```diff
@@ -11,6 +11,39 @@ def totals
     render(json: statistical_totals_json, status: 200)
   end
 
+  # Takes an array of channel identifiers and retuns a dictionary of channel identifiers as keys and true/false as values
+  def allowed_countries
+    channels = params[:channel_ids].map do |id|
+      {channel: Channel.find_by_channel_identifier(id), channel_identifier: id}
+    end
+
+    parameters = Rewards::Parameters.new.get_parameters
+
+    case parameters
+    when Rewards::Types::ParametersResponse
+      allowed_regions = parameters.custodianRegions
+    else
+      LogException.perform(parameters)
+      raise StandardError.new("Could not load allowed regions")
+    end
+
+    response = {}
+    channels.each do |channel_obj|
+      publisher = channel_obj[:channel].publisher
+
+      response[channel_obj[:channel_identifier]] = if publisher.uphold_connection.present?
```

Comment:
In the query, make sure you preload the items you need. You can use strict loading like [here](https://github.com/brave-intl/publishers/blob/fd712a808b18ac295462ff66f2816e70a35bc560/app/services/enqueue_publishers_for_payout_service.rb#L65)